File: src/main/java/com/domclick/model/Account.java
package com.domclick.model;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.PositiveOrZero;

@Entity
@Table(name = "BANK_ACCOUNT")
public class Account extends BaseEntity {
    @NotNull(message = "Баланс не может быть пустым")
    @PositiveOrZero(message = "Баланс не может быть отрицательным")
    @Column(name = "BALANCE", nullable = false)
    private Double balance;

    @ManyToOne
    @JoinColumn(name = "USER_ID", nullable = false)
    private User user;

    @Transient
    private Long userId;

    @Transient
    private String userStr;

    public Double getBalance() {
        return balance;
    }

    public void setBalance(Double balance) {
        this.balance = balance;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public String getUserStr() {
        return user.toString();
    }

    public void setUserStr(String userStr) {
        this.userStr = userStr;
    }

    public void updateUserId() {
        this.userId = user != null ? user.getId() : null;
    }
}

File: persistence/src/main/kotlin/com/domclick/entity/Account.kt
package com.domclick.entity

import java.math.BigDecimal
import javax.persistence.*
import javax.validation.constraints.NotNull
import javax.validation.constraints.PositiveOrZero

@Entity
@Table(name = "BANK_ACCOUNT")
//Todo use data classes
class Account(
        @NotNull(message = "Баланс не может быть пустым")
        @PositiveOrZero(message = "Баланс не может быть отрицательным")
        @Column(name = "BALANCE", nullable = false, precision = 19, scale = 2)
        var balance: BigDecimal? = BigDecimal(0),

        @ManyToOne
        @JoinColumn(name = "USER_ID", nullable = false)
        var user: User? = null
) : BaseEntity() {
    @Transient
    var userId: String = ""

    fun getUserStr() = user.toString()

    fun updateUserId() {
        this.userId = user!!.id.toString()
    }
}
